migrate_legacy_captures.js

```diff
@@ -0,0 +1,218 @@
+    growerAccount = await transaction("treetracker.grower_account").insert(
+      {
+        name: planter.first_name + " " + planter.last_name,
+        email: planter.email,
+        phone: planter.phone,
+        image_url: planter.image_url ?? legacyTree.planter_photo_url, //should this be public.trees.planter_photo_url?
+        image_rotation: planter.image_rotation ?? 0,
+        organization_id: planter.organization_id,
+        wallet_id: generate_uuid_v4(), // TODO: temporary workaround, needs to be a specific uuid and not a newly generated one.
+        wallet: "Temporary placeholder",
+        first_registration_at: new Date(),
+      },
+      [
+        "id",
+        "wallet_id",
+        "name",
+        "email",
+        "phone",
+        "organization_id",
+        "image_url",
+        "image_rotation",
+      ]
+    );
+  }
+  console.log(`GrowerAccount exists: ${JSON.stringify(growerAccount)}`);
+  return growerAccount;
+}
+
+async function migrate() {
+  try {
+    const base_query_string = `select t.* from public.trees t left join treetracker.capture c on t.id = c.reference_id
+                                where t.active=true and t.approved=true and t.image_url != null and
+                                c.reference_id is null`;
+    const rowCountResult = await knex.select(
+      knex.raw(`count(1) from (${base_query_string}) as src`)
+    );
+    console.log(`Migrating ${+rowCountResult[0].count} records`);
+
+    const bar = new ProgressBar("Migrating [:bar] :percent :etas", {
+      width: 20,
+      total: +rowCountResult[0].count,
+    });
+
+    const attributesFormatter = (attributes) => {
+      if (attributes.length <= 0) return null;
+      const attributes_in_json_format = { entries: [] };
+      for (let entry of attributes) {
+        attributes_in_json_format.entries.push(entry);
+      }
+      return attributes_in_json_format;
+    };
+
+    const createCapture = (
+      {
+        id,
+        time_created,
+        planter_photo_url,
+        gps_accuracy,
+        image_url,
+        lat,
+        lon,
+        morphology,
+        age,
+        note,
+        time_updated,
+      },
+      treeAttributes,
+      growerAccount
+    ) => {
+      return Object.freeze({
+        reference_id: id,
+        image_url: image_url,
+        lat: lat,
+        lon: lon,
+        gps_accuracy: gps_accuracy,
+        grower_id: growerAccount.id,
+        grower_photo_url: planter_photo_url,
+        grower_username: "-", //TODO: What should be used for username here?
+        morphology: morphology,
+        age: age,
+        note: note,
+        attributes: attributesFormatter(treeAttributes),
+        created_at: time_created,
+        updated_at: time_updated,
+        device_configuration_id: uuid_generate_v4(), // should this take value of device_identifier
+        session_id: uuid_generate_v4(), // TODO: Is this appropriate to generate uuid here?

```

Comment:
What should be used for session_id?

Reply:
session id is a dummy value.  This concept does not exist for legacy data.